transport/http/inbound.go

```diff
@@ -51,6 +51,15 @@ func Mux(pattern string, mux *http.ServeMux) InboundOption {
 	}
 }
 
+// Interceptor specifies a func that takes the transport http.Handler and
+// produces an http.Handler. This can be used to create custom behavior
+// when inbound HTTP requests are recieved.
```

Comment:
We should probably add a disclaimer that if people use this and **don't** call into the passed in handler, it is at their own risk.

Reply:
Maybe something like,

    Interceptor specifies a function which can wrap the YARPC handler. If
    provided, this function will be called with an http.Handler which will
    route requests through YARPC. The http.Handler returned by this function
    may delegate requests to the provided YARPC handler to route them through
    YARPC.

(Not 100% sure about that wording either, TBH.)